macros/base/end_print.cfg

```diff
@@ -7,16 +7,19 @@ gcode:
     {% set light_intensity_end_print = printer["gcode_macro _USER_VARIABLES"].light_intensity_end_print %}
     {% set ercf_enabled = printer["gcode_macro _USER_VARIABLES"].ercf_enabled %}
     {% set ercf_unload_on_end_print = printer["gcode_macro _USER_VARIABLES"].ercf_unload_on_end_print %}
+    {% set unload = params.UNLOAD_AT_END|default(0)|int %}
     {% set filter_enabled = printer["gcode_macro _USER_VARIABLES"].filter_enabled %}
     {% set light_enabled = printer["gcode_macro _USER_VARIABLES"].light_enabled %}
     {% set status_leds_enabled = printer["gcode_macro _USER_VARIABLES"].status_leds_enabled %}
     {% set bed_mesh_enabled = printer["gcode_macro _USER_VARIABLES"].bed_mesh_enabled %}
 
     PARK
 
-    {% if ercf_enabled and ercf_unload_on_end_print %}
-       # unload filament and park into ercf
-        ERCF_EJECT
+    {% if ercf_enabled %}
+        {% if unload|int == 1 or ercf_unload_on_end_print %}    # leaves the choice between slicer end print gcode or Klippain variables
```

Comment:
what takes precedence? slicer or config? with slicer i want to not unload the filament but the variable is defined for it.

Reply:
I thought that in this way it would allow the user either to leave more or less permanently the ejection at the end of the print (with your variable) or to allow it to be defined according to the profile of the printer in the slicer by putting `END_PRINT UNLOAD_AT_END=1` in the End G-code of the slicer

Reply:
As stated in another comment. I think it's needed to define a clear order and here it should be slicer priority: if END_PRINT `UNLOAD_AT_END` is defined, use it.
If it's not the case, then fallback to the variable.

Also please rename the variable to ERCF_UNLOAD_AT_END (to not confuse the users about it)

Reply:
I push an update... I can't test now. Are you OK?